include/motor-control/core/tasks/move_group_task.hpp

```diff
@@ -126,6 +124,8 @@ class MoveGroupTask {
     /**
      * Task entry point.
      */
+    template <motion_controller_task::TaskClient MotionControllerClient,
+              message_writer_task::TaskClient CanClient>
```

Comment:
I don't understand how & when this is getting called

Reply:
It gets called in `FreeRTOSTask` in the lambda in the `start` method, here: https://github.com/Opentrons/ot3-firmware/blob/5b1dd97a56ac7732d693c5926a819ff706a9dd6c/include/common/core/freertos_task.hpp#L49

The [std::apply](https://en.cppreference.com/w/cpp/utility/apply)+tuple pattern is the c++ way of doing a call like this in python:
```python

class Deferrer:
   def __init__(self, f, *defer_args, **defer_kwargs):
      self._f = f
      self._defer_args = defer_args
      self._defer_kwargs = defer_kwargs

  def run(self):
      self._f(*self._defer_args, **self._defer_kwargs)

def foo(a, b, c=1):
    print(f"{a} {b} {c}")

d = Deferrer(foo, 1, 2, c=34)
d.run() # prints "1, 2, 34"
```